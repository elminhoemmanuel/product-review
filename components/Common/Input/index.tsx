import React from 'react'

type InputProps = React.ComponentProps<'input'>

const Input = ({ ...rest }: InputProps) => {
    return (
        <div className=''>
            <input {...rest} className="text-sm w-full block p-3 text-ckdarkblue bg-transparent border 
            border-gray-200 hover:border-cklightblue rounded-md focus:outline-none placeholder-ckdarkblue"
            />
        </div>
    )
}

export default Input