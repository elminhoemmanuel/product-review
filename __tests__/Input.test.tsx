import Input from '@/components/Common/Input';
import { render, screen } from '@testing-library/react'

test("On render, check that Input field is present", () => {
    const { container } = render(
        <Input type="text" />
    );
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot()
})
