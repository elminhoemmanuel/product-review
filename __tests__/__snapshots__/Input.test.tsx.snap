// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On render, check that Input field is present 1`] = `
<div
  class=""
>
  <input
    class="text-sm w-full block p-3 text-ckdarkblue bg-transparent border border-gray-200 hover:border-cklightblue rounded-md focus:outline-none placeholder-ckdarkblue"
    type="text"
  />
</div>
`;
